packages/core/src/routes/connector.ts

```diff
@@ -292,29 +294,37 @@ export default function connectorRoutes<T extends AuthedRouter>(
     }),
     async (ctx, next) => {
       const {
-        params: { id },
+        params: { factoryId },
         body,
       } = ctx.guard;
       const { phone, email, config } = body;
 
       const subject = phone ?? email;
       assertThat(subject, new RequestError({ code: 'guard.invalid_input' }));
 
-      const connector = await getLogtoConnectorById(id);
+      const connectorFactories = await loadConnectorFactories();
+      const connectorFactory = connectorFactories.find(
+        ({ metadata: { id } }) => id === factoryId && !demoConnectorIds.includes(id)
+      );
       const expectType = phone ? ConnectorType.Sms : ConnectorType.Email;
 
       assertThat(
-        connector,
+        connectorFactory,
         new RequestError({
           code: 'connector.not_found',
           type: expectType,
+          factoryId,
         })
       );
-      assertThat(connector.type === expectType, 'connector.unexpected_type');
 
-      const { sendMessage } = connector;
+      assertThat(connectorFactory.type === expectType, 'connector.unexpected_type');
+
+      const { rawConnector } = await buildRawConnector(connectorFactory);
+
+      // Make sure the connector is passwordless and sendMessage method is available, refactor connector's generic type later.
+      isPasswordlessConnector(rawConnector);
```

Comment:
this should fail fast?

Reply:
actually the log is already covered by the code above?

Reply:
Agreed, but linter seems does not properly narrowed the type down, let me take a look and see if there is a better way to handle this.

Reply:
better to make the compiler/linter recognizes the correct type, not bypass or add unnecessary code